ipc/src/server.rs

```diff
@@ -172,7 +173,7 @@ impl<M: Metadata, S: Middleware<M>> ServerBuilder<M, S> {
 					})
 				})
 				.filter_map(|x| x)
-				.select(receiver.map_err(|e| {
+				.select_both(receiver.map_err(|e| {
```

Comment:
I for one would appreciate a comment as to why we're using `select_both` here – doesn't seem super-obvious! ;)